bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangScheduler.java

```diff
@@ -96,6 +101,25 @@ public static void workerCountDown() {
             workersDoneSemaphore.release();
         }
     }
+
+    private static void handleInterruptibleAfterExecution(WorkerExecutionContext ctx) {
+        if (ctx.interruptible && ctx.parent != null && ctx.parent.parent == null) {
```

Comment:
Just trying to understand, why we check "ctx.parent.parent == null". Please put documentation for that.